jsonrpc/src/main.ts

```diff
@@ -316,6 +318,18 @@ export namespace ConnectionStrategy {
 	}
 }
 
+export interface ConnectionOptions {
```

Comment:
Actually now that I looked through all the code a beter approach might be to add a cancellationStragey to the options. But default the stragegy is for message. Such a strategy would need to implement three things:

- create a token source for the server side
- cancel (e.g to send the message or create a file)
- do clean up work when the request has finished.

Reply:
what you describe is basically how client/server works. and the client constructs the folder path, and create/remove it. 

all other ones (server, protocol, jsonrpc) just use what the client has constructed. this I feel make who owns the folder very clear. pushing that down to jsonrpc makes it a bit hard to know who owns what.

for example, if jsonrpc got option saying file cancellation strategy and UUID for 2 connections, who creates those folder if it doesn't exist? it just assume that someone would have created it? then that someone should have known where the connection would look to find that folder.

if not, then both connection need to try to create the folder if it doesn't exist and try to delete the folder if it exists if connection is disconnected and that doesn't feel like as clear as client owning the folder since client is the only one that construct the path.

Reply:
I see the point. And I partly noticed this myself that is why at the end proposed the strategy. Then all of this will be incapsulated in there and will not leak into the JSON-RPC code. Client and servers can either share a strategy or have different once. I addition we could implement the file based cancellation outside of JSOB-RPC.